go-controller/pkg/ovn/egressgw.go

```diff
@@ -98,12 +104,139 @@ func (oc *DefaultNetworkController) addPodExternalGWForNamespace(namespace strin
+	// That way any remaining entries with labels having MACs set will get purged.
+	if len(validNextHopMACs) == 0 {
+		validNextHopMACs = append(validNextHopMACs, []byte("does-not-contain-anything"))
+	}
+	pods, err := oc.watchFactory.GetPods(namespace)
+	if err != nil {
+		return fmt.Errorf("unable to get pods from informer: %v", err)
+	}
+
+	var errs []error
+	for _, pod := range pods {
+		pod := pod
+		// Since it's executed in ovnkube-controller only for multi-zone-ic the following hack of filtering
+		// local pods will work. Error will be treated as best-effort and ignored
+		if localPod, _ := oc.isPodInLocalZone(pod); !localPod {
+			continue
+		}
+		podIPs, err := util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})
+		if err != nil && !errors.Is(err, util.ErrNoPodIPFound) {
+			errs = append(errs, fmt.Errorf("unable to fetch IP for pod %s/%s: %v", pod.Namespace, pod.Name, err))
+		}
+		for _, podIP := range podIPs { // flush conntrack only for UDP
+			// for this pod, we check if the conntrack entry has a label that is not in the provided allowlist of MACs
+			// only caveat here is we assume egressGW served pods shouldn't have conntrack entries with other labels set
+			err := util.DeleteConntrack(podIP.String(), 0, kapi.ProtocolUDP, netlink.ConntrackOrigDstIP, validNextHopMACs)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to delete conntrack entry for pod %s: %v", podIP.String(), err))
+			}
+		}
+	}
+	return utilapierrors.NewAggregate(errs)
+}
+
+func (oc *DefaultNetworkController) checkAndDeleteStaleConntrackEntries() {
+	namespaces, err := oc.watchFactory.GetNamespaces()
+	if err != nil {
+		klog.Errorf("Unable to get pods from informer: %v", err)
+		return
+	}
+	for _, namespace := range namespaces {
+		// flush here since we know we have added an egressgw pod and we also know the full list of existing gatewayIPs
+		existingGWs, err := oc.apbExternalRouteController.GetAdminPolicyBasedExternalRouteIPsForTargetNamespace(namespace.Name)
+		if err != nil {
+			klog.Errorf("Unable to retrieve gateway IPs for Admin Policy Based External Route objects for ns %s: %w", namespace.Name, err)
+			return
+		}
+		// by now the nsInfo cache must be repaired for this feature fully;
+		// however this introduces cache lock scale concerns by doing this every minute
+		// versus previously this was done purely using annotations
+		nsInfo, nsUnlock, err := oc.ensureNamespaceLocked(namespace.Name, false, nil)
+		if err != nil {
+			klog.Errorf("Failed to ensure namespace %s locked: %v", namespace, err)
+			return
+		}
+		for _, gwInfo := range nsInfo.routingExternalPodGWs {
+			existingGWs.Insert(gwInfo.gws.UnsortedList()...)
+		}
+		if config.OVNKubernetesFeature.EnableInterconnect && oc.zone != types.OvnDefaultZone {
+			existingGWs.Insert(nsInfo.routingExternalGWs.gws.UnsortedList()...)
+		}
+		nsUnlock()
+		if len(existingGWs) > 0 {
+			// TODO: ASK TROZET FOR LOCAL PODS ON THE NODE?!
```

Comment:
yes we should just filter our own node or if the pod is in local zone right?

Reply:
yes done in the `syncConntrackForExternalGateways` function, so will remove this comment